contracts/modules/STO/USDTieredSTO.sol

```diff
@@ -683,61 +626,50 @@ contract USDTieredSTO is USDTieredSTOStorage, STO, ReentrancyGuard {
      * @return uint256 Value in ETH or POLY
      */
     function convertFromUSD(FundRaiseType _fundRaiseType, uint256 _amount) public view returns(uint256) {
-        uint256 rate = getRate(_fundRaiseType);
-        return DecimalMath.div(_amount, rate);
+        return DecimalMath.div(_amount, getRate(_fundRaiseType));
     }
 
     /**
      * @notice Return the total no. of tokens sold
      * @return uint256 Total number of tokens sold
      */
-    function getTokensSold() external view returns (uint256) {
-        return _getTokensSold();
-    }
-
-    function _getTokensSold() internal view returns (uint256) {
+    function getTokensSold() public view returns (uint256) {
         if (isFinalized)
             return totalTokensSold;
-        else
-            return getTokensMinted();
+        return getTokensMinted();
     }
 
     /**
      * @notice Return the total no. of tokens minted
      * @return uint256 Total number of tokens minted
      */
-    function getTokensMinted() public view returns(uint256) {
-        uint256 tokensMinted;
+    function getTokensMinted() public view returns (uint256 tokensMinted) {
         for (uint256 i = 0; i < tiers.length; i++) {
             tokensMinted = tokensMinted.add(tiers[i].mintedTotal);
         }
-        return tokensMinted;
     }
 
```

Comment:
Optimization Suggestion?
I could be wrong here but is there a logic to using `tokensMinted` instead of `totalTokensSold`? Are they not essentially the same thing? `totalTokensSold` only seems to be update after the STO is finaliszed. Instead there currently is an additional function that regularly calculates `tokensMinted`.  If `totalTokensSold` was update as tokens were minted instead it would save the need for `getTokensMinted()`, which is called multiple times and recalculates each time. It could simply be replaced with `totalTokensSold`. This would then allow you to remove `getTokensMinted()` and also remove the `getTokensSold()` function as `totalTokensSold` already has a public getter.
So by adding `totalTokensSold = totalTokensSold.add(tierPurchasedTokens)` after `tierData.mintedTotal = tierData.mintedTotal.add(tierPurchasedTokens)` in the `_calculateTier` function I think it allows this optimization.
The `finalize()` function could then also be tidied up a bit.

Reply:
Thanks @F-OBrien - I've marked this as an issue to see where it can be cleaned up. The difference is meant to be the amount of tokens which were actually sold through the STO, vs. the number of tokens which have been minted (including minting back to the reserve wallet). As you say we could probably clean this up though. #535